app/src/main/java/com/simformsolutions/oneloop/ui/user/signup/signupphonevarification/SignupPhoneVerificationViewModel.kt

```diff
@@ -0,0 +1,118 @@
+package com.simformsolutions.oneloop.ui.user.signup.signupphonevarification
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.simform.navigation.Navigator
+import com.simform.navigation.core.getNavArgs
+import com.simformsolutions.oneloop.ui.user.navigation.SignupOtpVerification
+import com.simformsolutions.oneloop.ui.user.navigation.SignupPhoneVerification
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import javax.inject.Inject
+
+/**
+ * The ViewModel for [SignupPhoneVerificationScreen].
+ *
+ * Manages phone number verification with country code selection.
+ *
+ * @param navigator The [Navigator] instance used for navigation.
+ */
+@HiltViewModel
+class SignupPhoneVerificationViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val navigator: Navigator
+) : ViewModel() {
+
+    private val _uiState = MutableStateFlow(getDefaultUiState())
+    private val signupPhoneVerification: SignupPhoneVerification = requireNotNull(savedStateHandle.getNavArgs<SignupPhoneVerification>())
+    private val _countryCode = mutableStateOf("+1")  // Default country code
+    private val _isDialogVisible = mutableStateOf(false)  // Dialog visibility for country code selection
+    val isDialogVisible: State<Boolean> get() = _isDialogVisible  // Exposed state for dialog visibility
```

Comment:
From line number 32 to 36, Code needs to be cleaned up and I'm not sure why did you took separate states here. We have UI state for that matter.